src/pages/ProductDetails/ProductDetails.tsx

```diff
@@ -0,0 +1,257 @@
+                            style={{ backgroundColor: item.colorId }}
+                          >
+                            «
+                          </div>
+                        </div>
+                      ))}
+                  </div>
+                </div>
+                <div className="product-memory">
+                  <span>Select capacity</span>
+                  <div className="product-memory__block">
+                    {productDetails.storage.availableCapacity
+                      && productDetails.storage.availableCapacity.map(
+                        (item: any) => (
+                          <button
+                            key={item.name}
+                            type="button"
+                            onClick={() => setCapacity(item.name)}
+                            className={classNames('product-memory__item', {
+                              'product-memory__item_active':
+                                item.name === capacity,
+                            })}
+                          >
+                            <Link to={`/${productDetails.type}s/${item.link}`}>
+                              {item.name}
+                            </Link>
+                          </button>
+                        ),
+                      )}
+                  </div>
+                </div>
+                {productDetails && (
+                  <div className="product-options__price">{setPrice()}</div>
+                )}
+                <div className="product-options__buttons">
+                  {productId && productDetails && (
+                    <>
+                      <AddToCartBtn id={productId} color={currentColor} />
+                      <AddFavouriteBtn id={productId} />
+                    </>
+                  )}
+                </div>
+                <div className="tech-specs__block">
+                  {productDetails.display.screenSize && (
+                    <div className="tech-specs__item">
+                      <span>Screen</span>
+                      <p>{productDetails.display.screenSize}</p>
+                    </div>
+                  )}
+                  {productDetails.display.screenResolution && (
+                    <div className="tech-specs__item">
+                      <span>Resolution</span>
+                      <p>{productDetails.display.screenResolution}</p>
+                    </div>
+                  )}
+                  {productDetails.hardware.cpu && (
+                    <div className="tech-specs__item">
+                      <span>Processor</span>
+                      <p>{productDetails.hardware.cpu}</p>
+                    </div>
+                  )}
+                  {productDetails.storage.ram && (
+                    <div className="tech-specs__item">
+                      <span>RAM</span>
+                      <p>{productDetails.storage.ram}</p>
+                    </div>
+                  )}
+                </div>
```

Comment:
use function where you can get productDetails and will return only content of span depending on the condition

Reply:
https://mate-parttime.slack.com/archives/C04284Q7B4G/p1680111683876209